src/commands/diffHeadWithBranch.ts

```diff
@@ -0,0 +1,54 @@
+'use strict';
+import { CancellationTokenSource, TextEditor, Uri, window } from 'vscode';
+import { ActiveEditorCommand, Commands, getCommandUri } from './common';
+import { GlyphChars } from '../constants';
+import { Container } from '../container';
+import { Logger } from '../logger';
+import { Messages } from '../messages';
+import { BranchesAndTagsQuickPick, CommandQuickPickItem } from '../quickPicks';
+
+export class DiffHeadWithBranchCommand extends ActiveEditorCommand {
+
+    constructor() {
+        super([Commands.DiffHeadWithBranch]);
+    }
+
+    async execute(editor?: TextEditor, uri?: Uri): Promise<any> {
+        uri = getCommandUri(uri, editor);
+
+        let progressCancellation: CancellationTokenSource | undefined;
+
+        try {
+            const repoPath = await Container.git.getRepoPath(uri);
```

Comment:
There is some duplication right here with `diffDirectory.ts` (to open the modal to select a branch). 
Maybe we can do something here to mutualise code ?